src/Compilers/VisualBasic/Portable/CodeGen/EmitExpression.vb

```diff
@@ -1793,6 +1788,23 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGen
                     EmitPseudoVariableAddress(DirectCast(assignmentTarget, BoundPseudoVariable))
                     lhsUsesStack = True
 
+                Case BoundKind.Sequence
+                    Dim sequence = DirectCast(assignmentTarget, BoundSequence)
+
+                    If Not sequence.Locals.IsEmpty Then
+                        _builder.OpenLocalScope()
+
+                        For Each local In sequence.Locals
+                            Me.DefineLocal(local, sequence.Syntax)
+                        Next
+                    End If
+
+                    Me.EmitSideEffects(sequence.SideEffects)
+                    lhsUsesStack = EmitAssignmentPreamble(sequence.ValueOpt)
+
```

Comment:
C# frees the locals here and it could be a bug in C#. If sequence returns a field of a struct temp declared on the sequence, we may reuse the temp by the time we do the actual store.
Looks like a possible bug in C#.